### ISSUE TITLE
ServiceLoader thread safety issue

### ISSUE THREAD

**Issue by valb3r** [adorsys/xs2a-adapter#369]
I'm facing following exception when doing load tests:
```
aused by: de.adorsys.xs2a.adapter.service.exception.AdapterNotFoundException: Adapter with id [adorsys-integ-adapter] is not found

	at de.adorsys.xs2a.adapter.service.loader.AdapterServiceLoader.lambda$getAccountInformationService$0(AdapterServiceLoader.java:44)

	at java.util.Optional.orElseThrow(Optional.java:290)

	at de.adorsys.xs2a.adapter.service.loader.AdapterServiceLoader.getAccountInformationService(AdapterServiceLoader.java:44)

	at de.adorsys.xs2a.adapter.service.impl.AccountInformationServiceImpl.getAccountInformationService(AccountInformationServiceImpl.java:41)

	at de.adorsys.xs2a.adapter.service.impl.AccountInformationServiceImpl.createConsent(AccountInformationServiceImpl.java:36)

	at de.adorsys.opba.protocol.xs2a.service.xs2a.consent.Xs2aTransactionListConsentInitiate.doRealExecution(Xs2aTransactionListConsentInitiate.java:49)

	at de.adorsys.opba.protocol.xs2a.service.xs2a.consent.Xs2aTransactionListConsentInitiate.doRealExecution(Xs2aTransactionListConsentInitiate.java:22)

	at de.adorsys.opba.protocol.xs2a.service.ValidatedExecution.execute(ValidatedExecution.java:25)
```
As I can see it is caused by ServiceLoader accessed by multiple threads without synchronizartion.

I'm referring to
```
public <T extends AdapterServiceProvider> Optional<T> getServiceProvider(Class<T> klass, String adapterId) {
        MDC.put("adapterId", adapterId);

        ServiceLoader<T> serviceLoader = getServiceLoader(klass);

        return StreamSupport.stream(serviceLoader.spliterator(), false)
                   .filter(provider -> provider.getAdapterId().equalsIgnoreCase(adapterId))
                   .findFirst();
    }
```

As ServiceLoader class is not thread-safe:
https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html
Quote:
```
Instances of this class are not safe for use by multiple concurrent threads. 
```

Probable fix would be synchronizing getServiceProvider method.

**Comment by spetrychenko** [comment]
Hi @valb3r, fix already merged into develop. We're going to make a release next week.

**Comment by spetrychenko** [comment]
@valb3r The fix is available with the adapter version 0.0.8.
